### DISCUSSION TITLE
Potentially implementing argparse

### DISCUSSION THREAD

**Discussion by Denperidge** [fabiocaccamo/python-benedict#290] [Ideas]
So, I just accidentally [reinvented the wheel](https://github.com/Denperidge-Redpencil/AnyArgs/issues/7), but also less fancy than this wheel. 😅

For a different project I am working on, I would want this package mixed with Python's argparse to allow CLI commands. So hence my following question: is there interest in implementing argparse/CLI support? If so, I will try to at least kickstart it by checking if some code from my repo can be re-used into this project. If not, I will see whether to keep my code as is and perhaps implement Python-Benedict!

Have a nice day!
- Cat (she/they)

**Comment by fabiocaccamo** [comment]
@Denperidge it seems interesting! Could you show me an example of what would be possible to do with this feature?

**Reply by Denperidge** [reply]
Ofcourse! Sorry for the delay in response, here are two gifs which illustrate how it can be used/useful!

Example of usage:
![Example usage](https://github.com/fabiocaccamo/python-benedict/assets/27348469/7c8d3795-b2e5-4395-bb93-e37dad8bf470)


Example of help:
![Example help](https://github.com/fabiocaccamo/python-benedict/assets/27348469/608a032f-e079-433d-892e-2376b4179609)

**Comment by fabiocaccamo** [comment]
It's not really clear to me, could you describe the feature a bit please?

**Comment by Denperidge** [comment]
Essentially, it could allow the package to support configuration through the command line. It could be implemented as a from_cli function!

For example, instead of using { “username”: “Example” } with from_json,  users would be able to use script.py —username Example with from_cli, and get the exact same results!

**Reply by fabiocaccamo** [reply]
Now I understand, cool! Thank you for the explanation!

Yes a `from_cli` method that catches CLI arguments would be perfect, want you work on a PR for this?

**Comment by Denperidge** [comment]
No problem, and sure thing! I’ll have a PR up and running in the next few days or weeks

**Reply by fabiocaccamo** [reply]
Don't forget to add tests!

**Comment by Denperidge** [comment]
Excuse my wild incorrect estimation of when I could get around to this, but the PR is up! https://github.com/fabiocaccamo/python-benedict/pull/311

**Reply by fabiocaccamo** [reply]
Don't worry at all, thank you very much for contributing!
